lms/static/scripts/frontend_apps/components/dashboard/OrderableActivityTable.tsx

```diff
@@ -0,0 +1,65 @@
+import type { DataTableProps } from '@hypothesis/frontend-shared';
+import { DataTable } from '@hypothesis/frontend-shared';
+import { useOrderedRows } from '@hypothesis/frontend-shared';
+import type { OrderDirection } from '@hypothesis/frontend-shared/lib/types';
+import { useMemo, useState } from 'preact/hooks';
+
+import type { BaseDashboardStats } from '../../api-types';
+
+export type OrderableActivityTableProps<T extends BaseDashboardStats> = Pick<
+  DataTableProps<T>,
+  'emptyMessage' | 'rows' | 'renderItem' | 'loading' | 'title'
+> & {
+  columnNames: Partial<Record<keyof T, string>>;
+  initialOrderField: keyof T;
+};
+
+const statsColumns = ['last_activity', 'annotations', 'replies'] as const;
+
+export default function OrderableActivityTable<T extends BaseDashboardStats>({
+  initialOrderField,
+  rows,
+  columnNames,
+  ...restOfTableProps
+}: OrderableActivityTableProps<T>) {
+  const [order, setOrder] = useState<NonNullable<DataTableProps<T>['order']>>({
+    field: initialOrderField,
+    direction: 'ascending',
+  });
+  const orderedRows = useOrderedRows(rows, order);
+  const columns = useMemo(
+    () =>
+      Object.entries(columnNames).map(([field, label], index) => ({
+        field: field as keyof T,
+        label: label as string,
+        classes: index === 0 ? 'w-[60%]' : undefined,
+      })),
+    [columnNames],
+  );
+  const orderableColumns = useMemo(
+    () =>
+      (Object.keys(columnNames) as Array<keyof T>).reduce<
+        Partial<Record<keyof T, OrderDirection>>
+      >((acc, columnName) => {
+        // All stat columns should be ordered in descending order first
+        acc[columnName] = statsColumns.includes(columnName as any)
```

Comment:
`columnName` here is `keyof T`, which translates to `string | number | symbol`.

Without the casting to `any`, TypeScript throws `Type 'string' is not assignable to type '"last_activity" | "annotations" | "replies"'`.

I would expect that `includes` actually acts as a type guard and narrows down the type, hence, allowing this.